tests/functional_tests/make_test_signature.cc

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2019, The Monero Project
+// 
+// All rights reserved.
+// 
+// Redistribution and use in source and binary forms, with or without modification, are
+// permitted provided that the following conditions are met:
+// 
+// 1. Redistributions of source code must retain the above copyright notice, this list of
+//    conditions and the following disclaimer.
+// 
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list
+//    of conditions and the following disclaimer in the documentation and/or other
+//    materials provided with the distribution.
+// 
+// 3. Neither the name of the copyright holder nor the names of its contributors may be
+//    used to endorse or promote products derived from this software without specific
+//    prior written permission.
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+// THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+// THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include <stdio.h>
+#include "string_tools.h"
+#include "rpc/rpc_payment_signature.h"
+
+int main(int argc, const char **argv)
+{
+  if (argc > 2)
+  {
+    fprintf(stderr, "usage: %s <secret_key>\n", argv[0]);
+    return 1;
+  }
+
+  crypto::secret_key skey;
+
+  if (argc == 1)
+  {
+    crypto::public_key pkey;
+    crypto::random32_unbiased((unsigned char*)skey.data);
```

Comment:
It seemed more like a wallet thing. keypair was something I considered, but that's in cryptonote, not crypto, and I chose to keep cryptonote away from deps. keypair might make sense to move to crypto, except it's now got this device type...

Reply:
I don't understand, why not simply do this which doesn't have any additional dependency?
```diff
   if (argc == 1)
   {
     crypto::public_key pkey;
-    crypto::random32_unbiased((unsigned char*)skey.data);
-    crypto::secret_key_to_public_key(skey, pkey);
+    crypto::generate_keys(pkey, skey);
     printf("%s %s\n", epee::string_tools::pod_to_hex(skey).c_str(), epee::string_tools::pod_to_hex(pkey).c_str());
     return 0;
   }
```

Reply:
By "It seemed more like a wallet thing", I meant that it seemed like a function made specifically for wallets, since it's got the recovery system. But I'll change it, it seems like a good choice on second thought.